x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type.ts

```diff
@@ -0,0 +1,299 @@
+        { name: 'date', description: actionVariableContextDateLabel },
+        { name: 'value', description: actionVariableContextValueLabel },
+        { name: 'hits', description: actionVariableContextHitsLabel },
+        { name: 'conditions', description: actionVariableContextConditionsLabel },
+      ],
+      params: [
+        { name: 'index', description: actionVariableContextIndexLabel },
+        { name: 'esQuery', description: actionVariableContextQueryLabel },
+        { name: 'threshold', description: actionVariableContextThresholdLabel },
+        { name: 'thresholdComparator', description: actionVariableContextThresholdComparatorLabel },
+      ],
+    },
+    minimumLicenseRequired: 'basic',
+    executor,
+    producer: STACK_ALERTS_FEATURE_ID,
+  };
+
+  async function executor(
+    options: AlertExecutorOptions<
+      EsQueryAlertParams,
+      EsQueryAlertState,
+      {},
+      ActionContext,
+      typeof ActionGroupId
+    >
+  ) {
+    const { alertId, name, services, params, state } = options;
+    const previousTimestamp = state.latestTimestamp;
+
+    const callCluster = services.callCluster;
+    const { parsedQuery, dateStart, dateEnd } = getSearchParams(params);
+
+    const compareFn = ComparatorFns.get(params.thresholdComparator);
+    if (compareFn == null) {
+      throw new Error(getInvalidComparatorError(params.thresholdComparator));
+    }
+
+    // During each alert execution, we run the configured query, get a hit count
+    // (hits.total) and retrieve up to DEFAULT_MAX_HITS_PER_EXECUTION hits. We
+    // evaluate the threshold condition using the value of hits.total. If the threshold
+    // condition is met, the hits are counted toward the query match and we update
+    // the alert state with the timestamp of the latest hit. In the next execution
+    // of the alert, the latestTimestamp will be used to gate the query in order to
+    // avoid counting a document multiple times.
+
+    let timestamp: string | undefined = previousTimestamp;
+    const filter = timestamp
+      ? {
+          bool: {
+            filter: [
+              parsedQuery.query,
+              {
+                bool: {
+                  must_not: [
+                    { bool: { filter: [{ range: { [params.timeField]: { lte: timestamp } } }] } },
+                  ],
+                },
+              },
+            ],
+          },
+        }
+      : parsedQuery.query;
+
+    const query = buildSortedEventsQuery({
+      index: params.index,
+      from: dateStart,
+      to: dateEnd,
+      filter,
+      size: DEFAULT_MAX_HITS_PER_EXECUTION,
+      sortOrder: 'desc',
+      searchAfterSortId: undefined,
+      timeField: params.timeField,
+    });
+
+    logger.debug(`alert ${ES_QUERY_ID}:${alertId} "${name}" query - ${JSON.stringify(query)}`);
+
+    let searchResult: SearchResponse<unknown> = await callCluster('search', query);
+    // Needed until https://github.com/elastic/kibana/issues/26356 is resolved
```

Comment:
I thought we addressed this in https://github.com/elastic/kibana/issues/84222
Is this still blocked because we're still using the legacy `callCluster`? 🤔 
If so, best to include a comment regarding [this](https://github.com/elastic/kibana/issues/50247) issue instead

Reply:
Updated! Thanks for pointing me to that issue!